src/NServiceBus.Core/Sagas/PropertySagaFinder.cs

```diff
@@ -25,6 +25,15 @@ public override async Task<IContainSagaData> Find(IBuilder builder, SagaFinderDe
             var lookupValues = context.GetOrCreate<SagaLookupValues>();
             lookupValues.Add<TSagaData>(sagaPropertyName, propertyValue);
 
+            if (propertyValue == null)
+            {
+                var saga = context.Get<ActiveSagaInstance>();
+                var sagaEntityName = saga.Metadata.Name;
+                var messageName = finderDefinition.MessageTypeName;
+
+                throw new Exception($"Message {messageName} mapped to the saga of type {sagaEntityName} has attempted to assign null value to the correlation property {sagaPropertyName}. Correlation properties cannot be assigned null.");
```

Comment:
```suggestion
                throw new Exception($"Message {messageName} mapped to saga {sagaEntityName} has attempted to assign null to the correlation property {sagaPropertyName}. Correlation properties cannot be assigned null.");
```

Reply:
👍 The same change needs to happen in the test. Will apply in a single commit manually.